File: app/src/main/java/org/eldorado/simpleweather/data/remote/RemoteWeatherDataSource.java
package org.eldorado.simpleweather.data.remote;

import org.eldorado.simpleweather.data.model.Weather;

import retrofit2.Call;

public class RemoteWeatherDataSource {

    private final OpenWeatherService mOpenWeatherService;

    private RemoteWeatherDataSource(OpenWeatherService openWeatherService) {
        mOpenWeatherService = openWeatherService;
    }

    public static RemoteWeatherDataSource getInstance(OpenWeatherService openWeatherService) {
        return new RemoteWeatherDataSource(openWeatherService);
    }

    public Call<Weather> getCityWeather(String cityName) {
        return mOpenWeatherService.getCityWeather(
                cityName,
                RemoteContract.UNIT_METRIC,
                RemoteContract.ACCESS_KEY_API_LAYER
        );
    }
}

File: app/src/main/java/org/eldorado/simpleweather/data/remote/RemoteWeatherDataSource.kt
package org.eldorado.simpleweather.data.remote

class RemoteWeatherDataSource (private val openWeatherService: OpenWeatherService) {

    fun getCityWeather(cityName: String) =
            openWeatherService.getCityWeather(
                    cityName,
                    RemoteContract.UNIT_METRIC,
                    RemoteContract.ACCESS_KEY_API_LAYER
            )
}
